Gems/AtomLyIntegration/CommonFeatures/Code/Source/Material/EditorMaterialSystemComponent.cpp

```diff
@@ -240,7 +195,79 @@ namespace AZ
             m_materialPreviews.erase(entityId);
         }
 
-        void EditorMaterialSystemComponent::OnRenderMaterialPreviewComplete(
+        void EditorMaterialSystemComponent::OnSystemTick()
+        {
+            if (auto previewRenderer = AZ::Interface<AtomToolsFramework::PreviewRendererInterface>::Get())
+            {
+                static constexpr const char* DefaultModelPath = "models/sphere.azmodel";
+                static constexpr const char* DefaultLightingPresetPath = "lightingpresets/thumbnail.lightingpreset.azasset";
+
+                for (const auto& materialPreviewRequestPair : m_materialPreviewRequests)
+                {
+                    const auto& entityId = materialPreviewRequestPair.first;
+                    const auto& materialAssignmentId = materialPreviewRequestPair.second;
+
+                    AZ::Data::AssetId materialAssetId = {};
+                    MaterialComponentRequestBus::EventResult(
+                        materialAssetId, entityId, &MaterialComponentRequestBus::Events::GetMaterialAssetId, materialAssignmentId);
+                    if (!materialAssetId.IsValid())
+                    {
+                        return;
```

Comment:
It looks like this should _continue_ instead of _return_.

Reply:
Good catch! This was previously called in another function that handled one at a time.